### PR TITLE
Update to iojs, use native Promise

### PR DESCRIPTION
**Pull request by cpojer**
I'm using `then/promise` to shim the global Promise in the tests so that timers will continue to work and I won't have to rewrite all the tests. The tests pass but I'm not 200% sure all my changes are correct.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,6 @@
 {
   "name": "node-worker-pool",
-  "version": "2.4.5",
```
Comment (jeffmo) [inline package.json]: I usually leave this for a separate commit so I can use `npm version` (which will both bump the version, create a notable commit message in commit history as a linchpin, and create a tag -- all for me)

Comment (cpojer) [inline package.json]: that sounds really cool actually, I did not know

**Comment by cpojer** [comment]
Ok, removed the change to package.json

**Comment by jeffmo** [comment]
Merge away!

**Pull request merged**